### PR TITLE
Support global max concurrency

### PR DESCRIPTION
**Pull request by kargakis**
Depends on https://github.com/kubernetes/test-infra/pull/4712. Only the last commit is new.

Fixes https://github.com/kubernetes/test-infra/issues/4691
/area prow

### CODE REVIEW

File: prow/config/config.go [hunk 1]
```diff
@@ -97,6 +97,10 @@ type Plank struct {
 	// will be passed a kube.ProwJob and can provide an optional blurb below
 	// the test failures comment.
 	ReportTemplate *template.Template `json:"-"`
+
+	// MaxConcurrency is the maximum number of tests running concurrently that
+	// will be allowed by plank. 0 implies no limit.
+	MaxConcurrency int `json:"max_concurrency,omitempty"`
```
Comment (stevekuznetsov) [inline prow/config/config.go]: Still feel like we should embed a common struct for this stuff, not clear what we gain with the copy paste

File: prow/config/config.go [hunk 2]
```diff
@@ -218,6 +226,9 @@ func parseConfig(c *Config) error {
 		return fmt.Errorf("parsing template: %v", err)
 	}
 	c.Plank.ReportTemplate = reportTmpl
+	if c.Plank.MaxConcurrency < 0 {
+		return fmt.Errorf("plank has invalid max_concurrency (%d), it needs to be a non-negative number", c.Plank.MaxConcurrency)
```
Comment (stevekuznetsov) [inline prow/config/config.go]: Why not `uint`?

File: prow/config/config.go (hunk 1 above)
Comment (kargakis) [reply prow/config/config.go]: How would we use separate limits for Jenkins and k8s in that case?

File: prow/config/config.go (hunk 2 above)
Comment (kargakis) [reply prow/config/config.go]: Staying consistent with the ProwJob API and an API convention in Kube: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#primitive-types

> Do not use unsigned integers, due to inconsistent support across languages and libraries. Just validate that the integer is non-negative if that's the case.

File: prow/config/config.go (hunk 1 above)
Comment (stevekuznetsov) [reply prow/config/config.go]: I mean Go embedding

File: prow/config/config.go (hunk 2 above)
Comment (stevekuznetsov) [reply prow/config/config.go]: ack

File: prow/config/config.go (hunk 1 above)
Comment (kargakis) [reply prow/config/config.go]: Yeah, that's an option too but I would prefer to tackle it in a follow-up. Included in https://github.com/kubernetes/test-infra/issues/4744.

**Pull request merged**